rocketpy/mathutils/function.py

```diff
@@ -262,7 +262,7 @@ def set_source(self, source):
 
             # Finally set source, update extrapolation and interpolation
             self.source = source
-            self.set_extrapolation(extrapolation)
```

Comment:
If extrapolation changes, then `get_value_opt` has to change as well. Why would it not be needed here?